### PR TITLE
Feat: Add evals harness + improve skills

### PR DESCRIPTION
**Pull request by nieblara**

### CODE REVIEW

File: .github/workflows/eval-skills.yml [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+
+on:
+  pull_request:
+    paths:
+      - "skills/**"
+      - "evals/**"
+      - ".github/workflows/eval-skills.yml"
+  schedule:
+    # 09:17 UTC daily - off the hour to avoid lining up with API rate limits.
+    - cron: "17 9 * * *"
```
Comment (ari-launchdarkly) [inline .github/workflows/eval-skills.yml:23]: Is the rationale here (besides the rate limits) that we'd be available and not after-hours if this were to fire and trigger an alert? It's such a random time (i'm used to seeing Cron jobs running in the dead hours)

File: .github/workflows/eval-skills.yml [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+  workflow_dispatch:
+    inputs:
+      run_all:
+        description: "Re-run every suite regardless of diff"
```
Comment (ari-launchdarkly) [inline .github/workflows/eval-skills.yml:27]: are we sure about this? I think we'd only want to re-run these (since it can add up) when:

1. The temperature changes
2. A model changes
3. The prompt changes

There should be a way for us to detect that. If we want, it can be a fast-follow

Comment (ari-launchdarkly) [reply .github/workflows/eval-skills.yml:27]: Oh wait, I'm just realizing that this is to be run on the cron. Nevermind

File: .github/workflows/eval-skills.yml [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+      - name: Install eval dependencies
+        run: npm ci
+        working-directory: evals
+
+      - name: Run eval suite
+        env:
+          ANTHROPIC_API_KEY: ${{ secrets.ANTHROPIC_API_KEY }}
+          OPENAI_API_KEY: ${{ secrets.OPENAI_API_KEY }}
+          AGENT_MODEL: ${{ vars.AGENT_MODEL || 'claude-sonnet-4-20250514' }}
+          RUBRIC_MODEL: ${{ vars.RUBRIC_MODEL || 'anthropic:messages:claude-haiku-4-5-20251001' }}
```
Comment (ari-launchdarkly) [inline .github/workflows/eval-skills.yml:106]: this could just be me having limited permissions, but I don't see the OPENAI_API_KEY set in the repo secrets or the variables

Image
Image

File: .github/workflows/eval-skills.yml [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+        run: node scripts/aggregate.js --run --only=${{ matrix.suite }}
+        working-directory: evals
+
+      - name: Upload suite results
+        uses: actions/upload-artifact@v4
+        with:
+          name: results-${{ matrix.suite }}
+          path: evals/${{ matrix.suite }}/results.json
+          retention-days: 14
```
Comment (ari-launchdarkly) [inline .github/workflows/eval-skills.yml:115]: Do we care for outliers in the process that are found? I'm thinking from a data-ingest perspective where if we have failures over a period of time, could that provide an agent context on how to write better skills and evaluations as we iterate?

File: evals/aiconfig-create/promptfooconfig.yaml [hunk 5]
```diff
@@ -0,0 +1,267 @@
...
+        Project key is "support-app".
+      codebase_context: >
+        The codebase uses the LaunchDarkly Python AI SDK. AI Configs are evaluated
+        using ldclient with create_chat(). Config keys use kebab-case.
+    assert:
+      - type: javascript
+        value: |
+          const tools = output.tools_called || [];
+          const pass = tools.includes('setup-ai-config');
+          return { pass, score: pass ? 1 : 0, reason: tools.length ? 'Tools called: ' + tools.join(' -> ') : 'No tools called' };
```
Comment (ari-launchdarkly) [inline evals/aiconfig-create/promptfooconfig.yaml:36]: i'm a little confused by this. Should the `type` be javascript if we're setting up ldclient in a python app?

File: docs/evals.md [hunk 6]
```diff
@@ -0,0 +1,426 @@
...
+
+`evals/providers/claude-skill-agent-sdk.js` is the bridge between
+promptfoo's per-test loop and a running agent. For each suite:
+
+- It builds a per-skill isolated cwd at
+`evals/.tmp-skill-fixtures/<slug>/` containing only a symlink at
+`.claude/skills/<slug>/` back to the real skill source. The SDK's
+project-scoped skill discovery sees only that one skill, not every
+sibling in the repo.
+- It also redirects `CLAUDE_CONFIG_DIR` to an empty throwaway
```
Comment (ari-launchdarkly) [inline docs/evals.md:98]: do we need security to review this?

File: docs/evals.md [hunk 7]
```diff
@@ -0,0 +1,426 @@
...
+  `evals/tools/definitions.json` and has a mock in
+    `evals/mocks/tool-responses.json`. Add what's missing.
+3. Create `evals/<suite>/promptfooconfig.yaml` with 3-5 test cases:
+  happy path, variant input, exploration without context, edge case,
+    safety scenario. (See `evals/README.md` for the template.)
+4. Add the suite to `evals/scripts/_manifest.js` (`suite`, `skillKey`,
+  `skillDir`, `readme`).
+5. Add `eval:<suite>` and `eval:<suite>:single` scripts to
+  `evals/package.json` matching the pattern of the existing ones.
+6. Run `npm run eval:<suite>:single` to validate the pipeline, then
```
Comment (ari-launchdarkly) [inline docs/evals.md:383]: can we automate this?

File: docs/evals.md [hunk 8]
```diff
@@ -0,0 +1,426 @@
...
+  `npm run eval:all` (or `npm run eval:aggregate`) to refresh the
+    baseline.
+
+The aggregator + CI pick up the new suite automatically once it's in
+`_manifest.js`.
+
+## Open questions and known limitations
```
Comment (ari-launchdarkly) [inline docs/evals.md:390]: An additional thing to mention (I may have missed it) but there's a lot of intent management in agent-skills. What I mean by that is we seem to be creating a "voice" or identity for our agentic experience that the evaluations should capture.

File: evals/.env.example [hunk 9]
```diff
@@ -0,0 +1,25 @@
+# Copy to evals/.env and fill in. Both files are gitignored.
+#
+# REQUIRED: API key for both the agent (system under test) and the rubric
+# grader unless you switch the grader to a non-Anthropic provider via
+# RUBRIC_MODEL below.
+ANTHROPIC_API_KEY=
+
```
Comment (ari-launchdarkly) [inline evals/.env.example:7]: I have a feeling we'll be asked where to get this value from. We should link that here

File: evals/.env.example [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+# REQUIRED: the rubric grader for `llm-rubric` assertions. Wired into
+# shared/defaults.yaml as defaultTest.options.provider. Pick a cheaper model
+# than AGENT_MODEL since this only judges agent output and runs once per
+# rubric assertion.
+#
+# Examples:
+#   anthropic:messages:claude-haiku-4-5-20251001
+#   openai:gpt-5-mini
+#   openai:chat:gpt-4.1-mini
+RUBRIC_MODEL=anthropic:messages:claude-haiku-4-5-20251001
```
Comment (ari-launchdarkly) [inline evals/.env.example:22]: Same here

File: evals/README.md [hunk 11]
```diff
@@ -0,0 +1,390 @@
...
+### Model strategy
+
+Two distinct models, two env vars:
+
+| Variable | Used by | Default | Why |
+|----------|---------|---------|-----|
+| `AGENT_MODEL` | the provider (system under test) | `claude-sonnet-4-20250514` | Stays on Claude because that's representative of what users actually run. |
+| `RUBRIC_MODEL` | `defaultTest.options.provider` (rubric grader) | `anthropic:messages:claude-haiku-4-5-20251001` | Cheaper grader cuts cost roughly 10x without changing what's measured. |
+
+`EVAL_MODEL` (the legacy variable) is still honoured as a fallback for `AGENT_MODEL` so existing `.env` files keep working.
```
Comment (ari-launchdarkly) [inline evals/README.md:111]: should we support the legacy value or just remove it?

File: skills/ai-configs/aiconfig-create/SKILL.md [hunk 12]
```diff
@@ -114,9 +166,11 @@ If the user asks for more control or a step-by-step approach, use the individual
...
+3. `get-ai-config` -- verify the result (this call is **not** optional)
+
+**Execute all three steps in a single pass without stopping to ask for details.** Infer the variation key (`default`), name (`Default`), instructions/messages, and model from the user's request context. If the user asked for GPT-4o agent mode, you have enough to complete the entire flow. Only ask clarifying questions if the mode or model is truly ambiguous.
 
-**Execute all three steps without stopping to ask for details.** Infer the variation key (`default`), name (`Default`), instructions/messages, and model from the user's request context. If the user asked for GPT-4o agent mode, you have enough to complete the entire flow. Only ask clarifying questions if the mode or model is truly ambiguous.
+**Step 3 (the `get-ai-config` call) is mandatory regardless of how convincing the create response looks.** The two write tools may return what looks like a complete object, but only `get-ai-config` confirms the config was actually persisted with both the shell and variation linked. Skipping this step is a workflow violation — make the call even when you "feel" the previous responses already showed everything.
```
Comment (ari-launchdarkly) [inline skills/ai-configs/aiconfig-create/SKILL.md:173]: I'd love to get someone from AIC to verify these changes

**Review by ari-launchdarkly — COMMENTED** [review]
I'm still reviewing this. Just submitting it for now since the kids are awake and I've gotta make breakfast. I'd love to get someone from AIC to review the skill changes. So far so good - just some questions and asks to flesh some parts out. My thinking (and I could be wrong) is that non-devs will likely be more and more involved in this process and so we should orient the language to there? Maybe that's the wrong way of looking at it since an agent might be directing them, but if that's the case, maybe we need to provide the most amount of context for that agent?

File: evals/mocks/tool-responses.json [hunk 13]
```diff
@@ -0,0 +1,306 @@
...
+    "key": "{{toolKey}}",
+    "description": "{{toolDescription}}",
+    "schema": {
+      "type": "object",
+      "properties": {
+        "input": { "type": "string", "description": "Input value" }
+      }
+    }
+  }
+}
```
Comment (ari-launchdarkly) [inline evals/mocks/tool-responses.json:306]: how do we ensure there isn't drift between these mocks and the real APIs?

File: evals/onboarding-router/promptfooconfig.yaml [hunk 14]
```diff
@@ -0,0 +1,442 @@
...
+          out flags as a secondary surface to come back for.
+          Evaluate:
+          1. Did the agent pick ai-configs as the primary route (not flags)?
+          2. Did it acknowledge the flags request as a secondary surface they can come back for?
+          3. Did it briefly explain the choice (e.g. "I'll start with AI Configs since that's
+             the highest-leverage starting point for an AI app")?
+          4. Did it hand off cleanly without trying to do both at once?
+          Score 1.0 if all four are met, deduct 0.25 for each missed.
+        metric: precedence_quality
+        weight: 2
```
Comment (ari-launchdarkly) [inline evals/onboarding-router/promptfooconfig.yaml:442]: this file is awesome

**Review by ari-launchdarkly — APPROVED** [review]
Ok. My other concerns are keeping things in sync (a problem we can resolve later) and adding tests to some of the utility methods. They do a lot and it would be nice to see that we've worked out any potential edge cases here
